nco/custom.py

```diff
@@ -0,0 +1,268 @@
+    'byte': 'b',
+    'ubyte': 'ub',
+    'int8' : 'b',
+    'uint8' : 'ub',
+    'uint16': 'us',
+    'uint32': 'ui',
+    'int64': 'll',
+    'uint64': 'ull',
+    'string': 'sng'
+})
+
+npTypesNP = dict({
+    'float32' : np.float32, 
+    'float'   : np.float32 ,
+    'f'       : np.float32 ,
+    'float64' : np.float64,
+    'double'  : np.float64,
+    'd'       : np.float64,
+    'int32'   : np.int32,
+    'i'       : np.int32,
+    'l'       : np.int32,
+    'int16'   : np.int16,
+    's'       : np.int16,
+    'str'     : str,
+    'char'    : str,
+    'c'       : str,
+    'string'  : str,  
+    'sng'     : str,  
+    'byte'    : np.byte,
+    'b'       : np.byte,
+    'ubyte'   : np.ubyte,
+    'ub'      : np.ubyte,
+    'int8'    : np.byte,
+    'uint8'   : np.ubyte,
+    'uint16'  : np.uint16,
+    'us'      : np.uint16,
+    'uint32'  : np.uint32,
+    'u'       : np.uint32,
+    'ui'      : np.uint32,
+    'ul'      : np.uint32,
+    'int64'   : np.int64,
+    'll'      : np.int64,
+    'uint64'  : np.uint64,
+    'ull'     : np.uint64
+})
+
+
+
+__prog__ = os.path.splitext(os.path.basename(__file__))[0]
+
+
+class atted(object):
+
+    def __init__(self, mode='overwrite',attName="",varName="", Value=None,  sType=None,**kwargs):
+    #def __init__(self, **kwargs):
+        mode = kwargs.pop('mode', mode)
+        attName = kwargs.pop('attName', attName)
+        varName = kwargs.pop('varName', varName)
+        Value = kwargs.pop('Value', Value)
+        sType = kwargs.pop('sType',sType )
+            
+        if mode in validModes:
+           mode=validModes[mode]  
+        elif mode in validModes.keys():
+           pass
+        else:
+           raise Exception('{0}: mode "{1}" not found'.format("atted()","atted():",mode))
+        
+        if attName==None or len(attName)==0: 
+            raise Exception ('{0}: attName zero length\n'.format("atted()"))
+
+
+        # set member defaults 
+        self.mode=mode
+        self.attName=attName
+        self.varName=varName 
+        self.npType=None
+        self.npValue=None    
+
+        # dont bother about type & value
+        if self.mode=="d":
+            return None
+        
+        # deal with string quirk 
+        # if user specifies type 'string' or 'sng' with a single character string 
+        # the the output type should be 'sng and NOT 'c'. To do this we put the single string
+        # inside a list the the prnOptions() method treats it as it would a list of strings
+
+        if type(Value) is str and ( sType == 'string' or sType == 'sng'):  
+          Value=[Value]
+
+        # see if Value is iterable   
+        try:
+            if type(Value) is str:
+              raise TypeError("str is not iterable (for our purposes")
+            it=iter(Value)    
+            inputType=type(next(it))   
+            bIterable=True
+            del it   
+        except Exception:
+            bIterable=False 
+            inputType=type(Value) 
```

Comment:
be specific about which `Exception` you are catching. This is too general. Is this a `TypeError`?